bin/install.sh

```diff
@@ -565,6 +571,90 @@ function _setSDKVersion {
+  android_includes=$(android_arch_includes $arch)
+
+  local cflags=(-std=gnu89 -g -pedantic -I$root/build/uv/include -I$root/build/uv/src -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -D_LARGEFILE_SOURCE -fPIC -Wall -Wextra -Wno-pedantic -Wno-sign-compare -Wno-unused-parameter -Wno-implicit-function-declaration)
+  cflags+=("${android_includes[@]}")
+  local objects=()
+  local sources=("unix/async.c" "unix/core.c" "unix/dl.c" "unix/fs.c" "unix/getaddrinfo.c" "unix/getnameinfo.c" "unix/linux.c" "unix/loop.c" "unix/loop-watcher.c" "unix/pipe.c" "unix/poll.c" "unix/process.c" "unix/proctitle.c" "unix/random-devurandom.c" "unix/random-getentropy.c" "unix/random-getrandom.c" "unix/random-sysctl-linux.c" "unix/signal.c" "unix/stream.c" "unix/tcp.c" "unix/thread.c" "unix/tty.c" "unix/udp.c" fs-poll.c idna.c inet.c random.c strscpy.c strtok.c threadpool.c timer.c uv-common.c uv-data-getter-setters.c version.c)  
+
+  declare output_directory="$root/build/$arch-$platform/uv$d"
+  mkdir -p $output_directory
+
+  declare src_directory="$root/build/uv/src"
+  
+  trap onsignal INT TERM
+  local i=0
+  local max_concurrency=$CPU_CORES
+  build_static=0
+  declare static_library="$root/build/$arch-$platform/lib$d/libuv$d.a"
+
+  for source in "${sources[@]}"; do
+    if (( i++ > max_concurrency )); then
+      for pid in "${pids[@]}"; do
+        wait "$pid" 2>/dev/null
+      done
+      i=0
+    fi
+
+    declare object="${source/.c/$d.o}"
+    object=$(basename $object)
+    objects+=("$output_directory/$object")
+
+    {
+      if (( force )) || ! test -f "$output_directory/$object" || (( $(stat_mtime "$src_directory/$source") > $(stat_mtime "$output_directory/$object") )); then
+        mkdir -p "$(dirname "$object")"
+        echo "# compiling object ($arch-$platform) $(basename "$source")"
+        quiet $clang "${cflags[@]}" -c "$src_directory/$source" -o "$output_directory/$object" || onsignal
+        echo "ok - built $source -> $object ($arch-$platform)"
+        # Can't write back to variable in block, remove final library to force rebuild
+        rm $static_library 2>/dev/null
+      fi
+    } & pids+=($!)
+  done
+
+  for pid in "${pids[@]}"; do
+    wait "$pid" 2>/dev/null
+  done
+
+  if [ ! -f $static_library ]; then
+    build_static=1
+  fi
+  mkdir -p "$(dirname $static_library)"
+
+  if (( build_static )); then
+    quiet $ar crs "$static_library" "${objects[@]}"
+    if [ -f $static_library ]; then
+      echo "ok - built libuv ($arch-$platform): $(basename "$static_library")"
+    else
+      echo "failed to build $static_library"
+      exit 1
+    fi
+  
+  else
+    if [ -f $static_library ]; then
+      echo "ok - using cached static library ($arch-$platform): $(basename "$static_library")"
+    else
+      echo "static library doesn't exist after cache check passed: ($arch-$platform): $(basename "$static_library")"
+      exit 1
+    fi
+  fi
+
+  lib_size=$(stat_size $static_library)
+
+  if (( $lib_size < 800000 )); then
```

Comment:
same here, this check seems like it may be too pedantic

Reply:
See above. libuv.a is just over 800000 bytes for 2 of the android abis, the other 2 are larger.

Reply:
Resolved as discussed.